stdlib/public/Darwin/Foundation/NSError.swift

```diff
@@ -239,7 +239,11 @@ public func _getErrorDefaultUserInfo<T: Error>(_ error: T)
 
           switch key {
           case NSLocalizedDescriptionKey:
-            return (error as? LocalizedError)?.errorDescription
+            if let description = (error as? LocalizedError)?.errorDescription {
+                return description
+            }
+            // Fallback to error's description if we don't have any localized description.
+            return String(describing: error)
```

Comment:
Should this behavior be used for older OSs too? (See the `!hasUserInfoValueProvider` section below.)

Also, this needs a test case.

Reply:
> Should this behavior be used for older OSs too? (See the !hasUserInfoValueProvider section below.)

But this overlay will never ship to those OSes, right?

> Also, this needs a test case.

Yep, will add if the change is acceptable.

Reply:
> But this overlay will never ship to those OSes, right?

Oh, good point. Oh well.